elastalert/config.py

```diff
@@ -133,7 +135,9 @@ def load_options(rule, conf, args=None):
     except (KeyError, TypeError) as e:
         raise EAException('Invalid time format used: %s' % (e))
 
-    # Set defaults
+    # Set defaults, copy defaults from config.yaml
+    for key, val in base_config.items():
```

Comment:
Setdefault means it won't change the value if it already exists. ``rule.setdefault`` can never overwrite anything.